### PR TITLE
Use whitespace around operator definitions.

### PR DESCRIPTION
**Pull request by akirahrkw**
this PR is about "Use whitespace around operator definitions."
https://github.com/realm/SwiftLint/issues/60

thanks:)

### CODE REVIEW

File: Source/SwiftLintFramework/Location.swift [hunk 1]
```diff
@@ -48,7 +48,7 @@ Returns true if `lhs` Location is equal to `rhs` Location.
 
 :returns: True if `lhs` Location is equal to `rhs` Location.
 */
-public func ==(lhs: Location, rhs: Location) -> Bool {
+public func == (lhs: Location, rhs: Location) -> Bool {
```
Comment (jpsim) [inline Source/SwiftLintFramework/Location.swift]: :smile:

File: Source/SwiftLintFramework/Rules/OperatorFunctionWhitespaceRule.swift [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        }
+        var violations = [StyleViolation]()
+        if let nameOffset = flatMap(dictionary["key.nameoffset"] as? Int64, { Int($0) }),
+            let nameLength = flatMap(dictionary["key.namelength"] as? Int64, { Int($0) }),
+            let offset = flatMap(dictionary["key.offset"] as? Int64, { Int($0) }) {
+
+            let location = Location(file: file, offset: offset)
+            let startAdvance = advance(file.contents.startIndex, nameOffset)
+            let endAdvance = advance(startAdvance, nameLength)
+            let range = Range<String.Index>(start: startAdvance, end: endAdvance)
```
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/OperatorFunctionWhitespaceRule.swift]: `<String.Index>` should be inferred, is it not?

File: Source/SwiftLintFramework/Rules/OperatorFunctionWhitespaceRule.swift [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            let definition = file.contents.substringWithRange(range)
+
+            let ope1 = ["/", "=", "-", "+", "!", "*", "|", "^", "~", "?", "."].map({ "\\\($0)" })
```
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/OperatorFunctionWhitespaceRule.swift]: nice use of `map`. Can you use trailing closure syntax here?

**Comment by jpsim** [comment]
This is really nice work, @akirahrkw! I just had a few minor comments, let me know what you think, then we'll merge this in.

File: Source/SwiftLintFramework/Rules/OperatorFunctionWhitespaceRule.swift (hunk 2 above)
Comment (akirahrkw) [inline Source/SwiftLintFramework/Rules/OperatorFunctionWhitespaceRule.swift]: ah yes, this is inferred:) I will remove it

File: Source/SwiftLintFramework/Rules/OperatorFunctionWhitespaceRule.swift (hunk 3 above)
Comment (akirahrkw) [inline Source/SwiftLintFramework/Rules/OperatorFunctionWhitespaceRule.swift]: it means .map({"\\\($0)"}), doesn't it?
yeah I will use it.

**Comment by akirahrkw** [comment]
thanks for your feedback. I just pushed the change:)

**Comment by jpsim** [comment]
I made a few changes to your work in #83 to support multibyte characters & specify only a single space on each side of the operator. So I'm closing this in favor of #83. Very nice work on this, @akirahrkw!

**Comment by akirahrkw** [comment]
got it and thank you:)

**Pull request closed without merging**